Issue 534: Support for property paths in SHACL UI

smessie opened the pull request:
The document currently lacks screenshots but provides descriptions of the UI. We can add screenshots of potential UIs later, e.g., once we have a first implementation of a SHACL UI library.

Contributes to #534 

* [See this document rendered online here](https://raw.githack.com/w3c/data-shapes/issue-534-ui-property-paths/shacl12-ui/index.html)

smessie committed:
Define support for property paths in SHACL UI

bergos commented on shacl12-ui/index.html:
```diff
@@ -534,8 +539,265 @@ <h2>Core Constraints</h2>
+                        <div class="turtle">
+ex:PersonShape
+    a sh:NodeShape ;
+    sh:targetClass ex:Person ;
```
```suggestion
```

bergos commented on shacl12-ui/index.html:
```diff
@@ -534,8 +539,265 @@ <h2>Core Constraints</h2>
+
+ex:BookShape
+    a sh:NodeShape ;
+    sh:targetClass ex:Book ;
```
```suggestion
```

bergos commented on shacl12-ui/index.html:
```diff
@@ -534,8 +539,265 @@ <h2>Core Constraints</h2>
+                        <div class="turtle">
+ex:PersonShape
+    a sh:NodeShape ;
+    sh:targetClass ex:Person ;
```
```suggestion
```

bergos reviewed (changes requested):
I propose removing the targets from the examples. Only the section about target finding should contain them. The other sections should not need them, which makes the examples shorter and easier to grasp.

The rest looks good to me.

edmondchuc commented:
I’m wondering if we should mention the restrictions on alternative paths in edit mode as described in https://github.com/w3c/data-shapes/issues/534#issuecomment-3316912350, specifically on which predicate paths are allowed.

Should we also add a paragraph on the challenges of supporting complex paths in edit mode?

Otherwise, everything else looks good to me.

smessie committed:
Remove targets from the examples

Apply suggestions from code review

Co-authored-by: Thomas Bergwinkl <bergos@users.noreply.github.com>

smessie committed:
Add notes on restrictions and challenges

smessie commented:
@edmondchuc I’ve pushed a commit that addresses your comment. Is it what you had in mind, or are there specific things you’re missing, or is it too verbose? I’m open to suggestions.

bergos commented:
@smessie I think your last additions to alternative path and other complex paths are very useful.

One small thing: Can you add a shape to the *Other Complex Paths* example? I think it would be useful to show `sh:path` in that case.

smessie committed:
Add shape example to "Other Complex Paths" subsection

smessie commented:
@bergos I wasn't sure yet if a sequence path (`sh:path ( ex:hasPart ex:hasPart ) ;`) would maybe make more sense than the zero-or-more-path I've used now, but if you think it does, I can still change it.

bergos commented:
@smessie not required to change anything. The path is fine like it is.

TallTed commented:
Screenshots may still be overly ambitious, but wireframes may be helpful to comprehension.

edmondchuc commented on shacl12-ui/index.html:
```diff
@@ -534,8 +539,338 @@ <h2>Core Constraints</h2>
+        </p>
+        <p>
+            The following subsections outline the scenarios in which SHACL UI implementations
+            <strong>MUST</strong>, <strong>SHOULD</strong>, or <strong>MAY</strong> support different kinds of property
```
We might not need to state these since it's already mentioned in section 1.4.

smessie replied:
I kind of liked this just as an introduction of what is coming. It is not really defining the semantic meaning of the keywords as in 1.4. In fact, it even relies on that definition to introduce the subsequent subsections.

But if this is a strong opinion, I could remove the paragraph.

edmondchuc replied:
I agree that the paragraph is useful as an intro to the upcoming subsections. I was just wondering if mentioning the RFC2119 variations is necessary, since it feels a bit repetitive. No strong opinions either way! And I see the `<strong>` tags have been removed. I'll give it a tick now. Thanks!

smessie replied:
Aha, right, thanks for the clarification! I have removed the repetitive keywords now.

smessie committed:
Don't put RFC2119 keywords in bold

smessie committed:
Replace repetitive RFC2119 keywords
